op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -288,6 +303,24 @@ func runFullMigration(opts fullMigrationOptions) error {
 		return fmt.Errorf("old-db head block number not synced to the block immediately before the migration block number: %d != %d", head.Number.Uint64(), opts.migrationBlockNumber-1)
 	}
 
+	config, err := genesis.NewDeployConfig(opts.deployConfig)
+	if err != nil {
+		return err
+	}
+
+	// Verify that (migration-block-time and l1StartingBlockTag) or (l1-beacon-rpc and l1-beaconchain-url) are set.
+	switch {
+	case (opts.migrationBlockTime != 0) != (config.L1StartingBlockTag != nil):
+		// Check that either both migration block time and l1StartingBlockTag are set or unset.
+		return fmt.Errorf("if the migration-block-time flag is specified, the l1StartingBlockTag in the deploy config must also be specified and vice versa")
+	case (opts.l1BeaconRPC != "") != (opts.l1BeaconchainURL != ""):
+		// Check that either both l1BeaconRPC and l1BeaconchainURL are set or unset.
+		return fmt.Errorf("if the l1-beacon-rpc flag is specified, the l1-beaconchain-url must also be specified and vice versa")
+	case (opts.migrationBlockTime == 0) == (opts.l1BeaconRPC == ""):
+		// Check that either the migration block time and l1StartingBlockTag pair or the l1BeaconRPC and l1BeaconchainURL pair is set but not both.
+		return fmt.Errorf("either (migration-block-time and l1StartingBlockTag) or (l1-beacon-rpc and l1-beaconchain-url) flags must be specified")
+	}
+
 	log.Info("Source db is synced to correct height", "head", head.Number.Uint64(), "migrationBlock", opts.migrationBlockNumber)
```

Comment:
Nit: I  would move this up above the changes so it immediately follows the corresponding check